llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -1045,6 +1060,23 @@ bool LoopVectorizationLegality::canVectorizeInstrs() {
     }
   }
 
+  // Make sure that all compare instruction users are recurrent if in loop's BB.
+  if (MultiCmpsRed.size() > 0) {
```

Comment:
Why do we need similar loops/checks in both LoopVectorizationLegality.cpp and IVDecriptors.cpp?

Reply:
There might be other kind of reduction and compare instruction with multipe users, I have seen such a case. I think I will add a test for this case.